cmd/holo-files/internal/common/filebuffer.go

```diff
@@ -130,37 +140,25 @@ func (fb *FileBuffer) Write(path string) error {
 //the given FileBuffer contains file contents, the same buffer is returned
 //unaltered.
 //
-//It uses the FileBuffer's BasePath to resolve relative symlinks. Since
-//file buffers are usually written to the target path of a `holo apply`
-//operation, the BasePath is most likely the target path.
-func (fb *FileBuffer) ResolveSymlink() (*FileBuffer, error) {
+//It uses the FileBuffer's Path to resolve relative symlinks.
+func (fb FileBuffer) ResolveSymlink() (FileBuffer, error) {
 	//if the buffer has contents already, we can use that
-	if fb.Contents != nil {
+	if fb.Mode&os.ModeSymlink == 0 {
 		return fb, nil
 	}
 
 	//if the symlink target is relative, resolve it
-	target := fb.SymlinkTarget
+	target := fb.Contents
 	if !filepath.IsAbs(target) {
-		baseDir := filepath.Dir(fb.BasePath)
+		baseDir := filepath.Dir(fb.Path)
 		target = filepath.Join(baseDir, target)
 	}
 
-	//read the contents of the target file (NOTE: It's tempting to just use
-	//NewFileBuffer here, but that might give us another FileBuffer with a
-	//symlink in it, and this time the symlink target might not resolve
-	//correctly against the original BasePath. So we explicitly read the file.)
-	contents, err := ioutil.ReadFile(target)
-	if err != nil {
-		return nil, err
-	}
-	return NewFileBufferFromContents(contents, fb.BasePath), nil
+	return newFileBuffer(target, true)
 }
 
 //EqualTo returns whether two file buffers have the same content (or link target).
-func (fb *FileBuffer) EqualTo(other *FileBuffer) bool {
-	if fb.Contents != nil {
-		return bytes.Equal(fb.Contents, other.Contents)
-	}
-	return fb.SymlinkTarget == other.SymlinkTarget
+func (fb FileBuffer) EqualTo(fa FileBuffer) bool {
+	fb.Path = fa.Path
+	return fa == fb
```

Comment:
`fb` and `fa` are passed by value, not by reference.  Changing `fb.path` doesn't propagate to outside of the function.